db/fetcher/fetcher.go

```diff
@@ -656,22 +656,18 @@ func (df *DocumentFetcher) FetchNextDoc(
 
 // Close closes the DocumentFetcher.
 func (df *DocumentFetcher) Close() error {
-	if df.kvIter == nil {
-		return nil
-	}
-
-	err := df.kvIter.Close()
-	if err != nil {
-		return err
-	}
-
-	if df.kvResultsIter == nil {
-		return nil
+	if df.kvIter != nil {
```

Comment:
Should these iterators be set to `nil` after closing to avoid calling `Close` again?

Reply:
Is an interesting thought, but am leaning towards 'I don't think so'.  I see it as the norm for Close func to assume that they will only ever be called once, and whilst you could argue that setting stuff to nil and allowing the GC to dispose of the objects is within the scope of a func call, it still feels slightly off to me.

Have you seen such things done much before?